Jenkinsfile

```diff
@@ -69,6 +71,18 @@ for(i = 0; i < buildTypes.size(); i++) {
 builds.failFast = failFast
 parallel builds
 
+if (buildTypes.contains("linux")) {
+    node("linux") {
+        unstash "metadata"
+        unarchive mapping: ['war/target/linux-jenkins.war': 'jenkins.war']
```

Comment:
This is not very nice from the perspective of minimizing overall build time, because the acceptance tests cannot begin until all of the functional tests—on both Linux and Windows—have completed. We really want ATH to run concurrently, as a branch in `builds`, but `jenkins.war` is not available from the start.

One solution would be to make the `linux` branch of the main build internally use `parallel` to set a Groovy flag once `war/target/jenkins.war` has been created; this is fairly early on, since to that point only unit tests have run. `waitUntil` allows this kind of cross-branch synchronization with some awkwardness, pending [JENKINS-27127](https://issues.jenkins-ci.org/browse/JENKINS-27127). But easier would be to simply do something along the lines of

```groovy
builds.ath = {
  node('linux') {
    checkout scm
    sh 'mvn -DskipTests -am -pl war package'
    runATH jenkins: "file:${pwd()}/war/target/jenkins.war", metadataFile: "${pwd()}/essentials.yml"
  }
}
```

After all, the build without tests is pretty cheap, so we are not wasting a lot of CPU repeating it.

Reply:
> We really want ATH to run concurrently, as a branch in builds

If that is the case I have no problem in changing it the way you describe, I used the non concurrent approach as it was the one used in #3073